src/io/socket.rs

```diff
@@ -206,6 +206,18 @@ impl Socket {
         std::mem::forget(s);
         result
     }
+
+    pub fn set_nodelay(&self, nodelay: bool) -> io::Result<()> {
+        use std::os::unix::io::FromRawFd;
+
+        let fd = self.as_raw_fd();
+
+        // SAFETY: See note for Self::shutdown
+        let s = unsafe { socket2::Socket::from_raw_fd(fd) };
```

Comment:
Use SockRef

Reply:
Done! You could probably make the same change for `shutdown ` as well.

Reply:
Yeah, I forgot it was done like that.